pandas/core/frame.py

```diff
@@ -8116,9 +8117,14 @@ def _series_round(s, decimals):
     # ----------------------------------------------------------------------
     # Statistical methods, etc.
 
-    def corr(self, method="pearson", min_periods=1) -> "DataFrame":
+    def corr(
+        self,
+        method: Union[str, Callable[[np.ndarray, np.ndarray], np.float64]] = "pearson",
```

Comment:
can you move this signature to an alias and put it in pandas._typing, cal it MethodWithCallable